sound/soc/sof/ipc4-priv.h

```diff
@@ -83,6 +83,20 @@ struct sof_ipc4_fw_data {
 			    struct sof_ipc4_fw_library *fw_lib, bool reload);
 };
 
+/**
+ * struct sof_ipc4_timestamp_info - IPC4 timestamp info
+ * @host_copier: the host copier of the pcm stream
+ * @dai_copier: the dai copier of the pcm stream
+ * @stream_start_offset: reported by fw in memory window
+ * @llp_offset: llp offset in memory window
+ */
+struct sof_ipc4_timestamp_info {
+	struct sof_ipc4_copier *host_copier;
+	struct sof_ipc4_copier *dai_copier;
+	u64 stream_start_offset;
+	u32 llp_offset;
```

Comment:
@RanderWang may I ask if this "llp_offset" is FW-specific or if this is platform-specific. Should we consider moving the delay calculation to the platform specific code or should it be in the IPC4 common code?

Reply:
@ranj063 llp_offset is applicable to CAVS and ACE platforms, so  it is FW-specific.